### PR TITLE
fix: allow to run codegen outside GOPATH

### PR DESCRIPTION
**Pull request by agaudreault**
When the project is not checked out in the standard go path, the local codegen does not work and running inside a container takes a lot of time. This PR creates a symlink in the GOPATH if the project is not checked out there and allows the codegen to run normally. It uses an explicit parameter even if having GOPATH exported in the script would work.

Tested with all the local codegen. 

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [x] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [x] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [x] I have added a brief description of why this PR is necessary and/or what this PR solves.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -192,7 +192,10 @@ gogen: ensure-gopath
 	go generate ./util/argo/...
 
 .PHONY: protogen
-protogen: ensure-gopath mod-vendor-local
+protogen: mod-vendor-local protogen-fast
```
Comment (blakepettersson) [inline Makefile:195]: Great stuff! Would it be worth adding `install-go-tools-local` to the task as well? It's not so clear that this is required for this to work.

Comment (agaudreault-jive) [reply Makefile:184]: The doc specifies to setup the local toolchain with `make install-tools-local` which will do the installation. I think this is enough and I do not really like having heavy steps always executed on make.

**Review by blakepettersson — APPROVED** [review]
LGTM

**Review by ishitasequeira — APPROVED** [review]
LGTM!!

**Pull request merged**